lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -371,26 +372,27 @@ sub _extended_needle_info {
 
 sub src {
     my ($self) = @_;
-
     return $self->reply->not_found unless $self->init();
 
     my $job    = $self->stash('job');
     my $module = $self->stash('module');
 
+    if (my $casedir = $job->settings->find({key => 'CASEDIR'})) {
+        my $casedir_url = Mojo::URL->new($casedir->value);
+        # if CASEDIR points to a remote location let's assume it is a git repo
+        # that we can reference like gitlab/github
+        last unless $casedir_url->scheme;
+        $casedir_url->path($casedir_url->path . '/blob/' . $casedir_url->fragment . '/' . $module->script);
+        $casedir_url->fragment('');
+        return $self->redirect_to($casedir_url);
```

Comment:
I wouldn't say placing a redirection is generally a bad idea but have you considered to render the desired link in the first place? Why redirecting when we can provide the desired link in the first place?

Reply:
1. To make it more obvious to users
2. Didn't know how to render the link in-place. Tried `$self->render_to_string` and `$self->render(text => …)`

Reply:
What does rendering in-place mean?

Reply:
Well basically I would prefer to have the URL directly put where "src" is called, i.e. https://github.com/os-autoinst/openQA/blob/master/templates/webapi/test/module_table.html.ep#L22 . How can I achieve that?

Reply:
Oh, i see. If you already know all the parameters in the template you could turn this whole action into a helper for the template that returns a link.

Reply:
The code can keep using the `link_to` helper. Make the URL itself preferably within the controller, e.g. at the point where `$module` from the line you've linked is initialized. You can likely not use `url_for` for external links (so just don't use it).

Reply:
not sure I understand. How can I tell openQA to call the method "src" then if not by `url_for`?

Reply:
My thought was to simply avoid the indirection of a separate external request. And to turn the `src` action into a helper. Like:
```
# In one of our helper plugins
$app->helper(src_link_for => sub {
  my ($c, $name, %args) = @_;
  ... do the stuff the src action does now...
  return $c->helpers->link_to(...);
});
```
And have the template call:
```
%= src_link_to $module->{name}, stepid => 1, moduleid => $module->{name}, testid => $testid
```

Reply:
yes, the helper part I understand and I can do that. I did not get how mkittler envisioned it. Thanks for the detailed suggestion. Saves me same wrong turns :)